### DISCUSSION TITLE
Slow build process when moving from Razzle 3.1.3 to 4.0.4

### DISCUSSION THREAD

**Discussion by asundlihardig** [jaredpalmer/razzle#1635] [Q&A]
We are in the process of moving our huge app from Razzle 3.1.3 to 4.0.4 

Everything is working really well and we are quite happy. Only one problem: Build speed for production.

Heroku kills the process because it takes too long. It's waiting forever on "Compiling client default build..."

Recently the local build gets killed after 1.5 minutes, telling us: JavaScript heap out of memory

We are using @loadable for code splitting.

Are there anything we can do to speed things up?

**Comment by fivethreeo** [comment]
The build can hang if there are errors in razzle.config.js

**Reply by asundlihardig** [reply]
Thank you. I’ll check that. We added one resolve could causing this. Nice😱

Which version of webpack should we use? 4 or 5?

**Reply by fivethreeo** [reply]
I would use 5 unless you want to use the experimental server side components, lots of improvements in 5. Also could add some options to allow increasing heap size in node. Norsk jo 😀

**Reply by fivethreeo** [reply]
May be related? https://github.com/jaredpalmer/razzle/issues/1587

**Reply by fivethreeo** [reply]
Got it working?

**Reply by asundlihardig** [reply]
I'm still trying to make it work

**Reply by fivethreeo** [reply]
Releasing 4.0.5 now, should report razzle.config.js errors.

**Reply by asundlihardig** [reply]
🙌🏻

**Reply by fivethreeo** [reply]
https://github.com/facebook/create-react-app/issues/8096

**Reply by fivethreeo** [reply]
This still happens locally?

**Reply by fivethreeo** [reply]
Make sure any large libraries are lazy loaded by loadable. May be too large a graph for webpack to keep in memory.
